src/v/redpanda/application.cc

```diff
@@ -664,14 +665,27 @@ void application::wire_up_services() {
           _proxy_client,
           to_yaml(*_proxy_client_config, config::redact_secrets::no))
           .get();
+
+        construct_single_service(_proxy_client_cache);
+
+        _proxy_client_cache
+          ->start(
+            smp_service_groups.proxy_smp_sg(),
+            to_yaml(*_proxy_client_config, config::redact_secrets::no),
+            _proxy_config->client_cache_max_size.bind(),
+            _proxy_config->client_keep_alive.bind())
```

Comment:
i think here is where you'd pass in lambdas or the sharded_argument wrapper and grab the binding on each core.